plugin/core/types.py

```diff
@@ -236,6 +219,14 @@ def r(name: str, default: Union[bool, int, str, list, dict]) -> None:
             r("inhibit_snippet_completions", False)
             r("inhibit_word_completions", True)
 
+        # Backwards-compatible with "diagnostics_highlight_style"
+        diagnostics_highlight_style = s.get("diagnostics_highlight_style")
+        if isinstance(diagnostics_highlight_style, str) and not s.has("show_diagnostics_highlights"):
+            if diagnostics_highlight_style:
+                self.show_diagnostics_highlights = True
+            else:
+                self.show_diagnostics_highlights = False
```

Comment:
Does this code sees only user overrides or the final state of the settings, including default setting? Because in the latter case the `has` check will not work as expected.

What I meant is that we could do:

```suggestion
        if isinstance(diagnostics_highlight_style, str):
            if not diagnostics_highlight_style:
                self.show_diagnostics_highlights = False
```

Since the fact that `show_diagnostics_highlights` is set to non-empty value shouldn't force the diagnostics to be enabled, even if `show_diagnostics_highlights` is `false`. The user overrode it to indicate that a different style is to be used but if he also adds `show_diagnostics_highlights: false` then that should win.

Reply:
That said, your added `s.has("show_diagnostics_highlights")` I guess makes sense also but does it work as expected though?

Reply:
You're probably right about `has` always being true